### PR TITLE
Support for required options

### LINKED ISSUE #203: support required arguments
if a required argument is not passed in, the program should error.
e.g.
```bash
% ruby foo.rb --arg blah
...output...
% ruby foo.rb
required argument --arg not given
%
```

### PR DESCRIPTION
**Pull request by woodruffw**
I've been needing this in one of my own projects and I saw that someone previously requested it in #203, so I went ahead and implemented it :smile: 

Let me know if you'd like me to update the README and/or Gem version in this PR as well.

Thanks for the awesome library!

Fixes #203

### CODE REVIEW

File: lib/slop/parser.rb [hunk 1]
```diff
@@ -80,6 +80,13 @@ def parse(strings)
 
       @arguments += ignored_args
 
+      unused_options.each do |o|
+        if o.config[:required] && !suppress_errors?
```
Comment (olleolleolle) [inline lib/slop/parser.rb]: Perhaps the whole loop can be obviated when `!suppress_errors?`

Comment (woodruffw) [reply lib/slop/parser.rb]: Very true. I'll fix that.

**Comment by olleolleolle** [comment]
This is a cool change. Can you teach the `required: true` option somehow, in some text file or the like?

**Comment by woodruffw** [comment]
> Can you teach users the required: true option somehow, in some text file or the like?

Sure thing. Should I add it to the README, or is somewhere else better?

**Comment by olleolleolle** [comment]
@woodruffw README's best!

**Comment by olleolleolle** [comment]
(To fully answer: gem version should not be changed with a feature. That's a separate release acitivity.)

**Comment by woodruffw** [comment]
Updated the README, plus I added `Option#required?` for consistency with `Option#suppress_errors?` and `Option#default_value`.

> (To fully answer: gem version should not be changed with a feature. That's a separate release acitivity.)

:+1:

**Comment by woodruffw** [comment]
Any update on this @olleolleolle?

**Comment by woodruffw** [comment]
Thanks! Would it be possible to get a version released for this change? I'd love to start using it :smile:

**Comment by leejarvis** [comment]
Thanks both. I've released 4.6.0

**Pull request merged**